### DISCUSSION TITLE
Sample ESP IDF code to change BLE adversting name during run time.

### DISCUSSION THREAD

**Discussion by RuchaVs** [community/community#77169] [Programming Help]
### Select Topic Area

Question

### Body

hello,
I have problem statement where the bluetooth  Adverstising name needs to be changed during Runtime.
 i am aware that bluetooth name can be changed through  esp_ble_gap_set_device_name, but this function is accepting a pointer to a fixed string whereas i am going to receive the new bluetooth name in a unsigned char buffer.

So i need a sample code to implement changing BLE adversting name during runtime,where the function will accept the new name input stored in a char buffer instead of a cont char pointer.

**Comment by RuchaVs** [comment]
Hello,

could please clarify what does it mean by write state alo.


Thanks,

________________________________

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @RuchaVs ! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the GitHub product Discussions. 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.
